extension/chrome/settings/modules/contacts.ts

```diff
@@ -141,7 +142,7 @@ View.run(class ContactsView extends View {
     const email = $('#edit_contact .input_pubkey').attr('email');
     if (!armoredPubkey || !email) {
       await Ui.modal.warning('No public key entered');
-    } else if (await PgpKey.fingerprint(armoredPubkey)) {
+    } else if (await PgpKey.fingerprint(await KeyUtil.parse(armoredPubkey))) {
```

Comment:
I wonder about code like this - there are quite a few similar usages. The intention of the original code is that some OpenPGP keys - when we try to parse them - return a null for fingerprint. But the type of `Key.id` is not optional, so I wonder where would such null go. Maybe it would throw earlier? Maybe the `.id` would actually contain a null, even if the type says it doesn't? This would be something to investigate, plus make sure we are testing this particular code branch.

Reply:
:thinking: do you have such a key for tests? I'd like to see how would the app behave.

If we need to support them then yeah `id` needs to be nullable. If not, I'd rather throw on parse rather to check for nulls in the entire codebase...

Reply:
I'd also rather throw early. I don't have any such key now (I may have encountered this before starting to write tests..) but as long as it does throw early, that's cool with me - there is nothing useful we can do with such keys.